offchain-modules/packages/x/src/utils/index.ts

```diff
@@ -118,6 +119,19 @@ export function privateKeyToCkbAddress(privkey: string, prefix: ckbAddressPrefix
   return utils.privateKeyToAddress(privkey, { prefix: prefix as AddressPrefix });
 }
 
+// since there may be many different formats of ckb address for the same lockscript,
+// we have to compare them after parsing to lockscript
+// return true if they are the same lockscript
+export function compareCkbAddress(address1: string, address2: string): boolean {
+  const lockscript1 = parseAddress(address1);
+  const lockscript2 = parseAddress(address2);
+  return (
+    lockscript1.code_hash === lockscript2.code_hash &&
+    lockscript1.args === lockscript2.args &&
```

Comment:
i found that if someone uses eth addr as an args from MetaMask that will make the args like this `0x6C6c6CCc...`, so maybe we may need to pay attention to the case